cli/flox/src/commands/environment.rs

```diff
@@ -570,25 +612,171 @@ impl Activate {
             },
         };
 
-        if !self.run_args.is_empty() {
+        if !run_args.is_empty() {
             command.arg("-i");
             command.arg("-c");
-            command.arg(self.run_args.join(" "));
+            command.arg(run_args.join(" "));
         }
 
         debug!("running activation command: {:?}", command);
 
-        if self.run_args.is_empty() {
+        if run_args.is_empty() {
             let message = formatdoc! {"
                 ✅  You are now using the environment {now_active}.
                 To stop using this environment, type 'exit'"};
             info!("{message}");
         }
-        let error = command.exec();
-
         // exec should never return
+        command.exec().into()
+    }
+
+    /// Patch the PATH to undo the effects of `/usr/libexec/path_helper`
+    ///
+    /// Darwin has a "path_helper" which indiscriminately reorders the path
+    /// to put the system curated path items first in the `PATH`,
+    /// which completely breaks the user's ability to manage their `PATH` in subshells,
+    /// e.g. when using tmux.
+    ///
+    /// On macos `/usr/libexec/path_helper` is typically invoked from
+    /// default shell rc files, e.g. `/etc/profile` and `/etc/zprofile`.
```

Comment:
nonblocking: it would help me understand this if the comment had an example order of events. Why is `/etc/profile` getting sourced after the `flox activate`?

Reply:
Added an example

Reply:
Thanks